codebuild/README.md

```diff
@@ -0,0 +1,926 @@
+### Step 5: Test self-hosted runner with a wokflow file
+
+Create a test github workflow file to test your self hosted runner. You can use the sample test-new-runner.yml file in the test-module directory.
+
+NOTE: Don't forget to update your runs-on: with the output lable i.e. codebuild-demo-runner-dev-runner-${github.run_id}-${github.run_attempt}
+
+```yaml
+
+name: Test New CodeBuild Runner
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - *
+    paths:
+      - 'test-module/**'
+      - 'terraform-aws-codebuild-github-runner/**'
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  test-runner:
+    name: Test New Runner
+    runs-on: codebuild-demo-runner-dev-runner-${github.run_id}-${github.run_attempt}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Test AWS Credentials
+        run: |
+          echo "Testing AWS credentials..."
+          aws sts get-caller-identity
+
+      - name: Test Environment
+        run: |
+          echo "Testing runner environment..."
+          echo "Runner: ${{ runner.name }}"
+          echo "OS: ${{ runner.os }}"
+          echo "Architecture: ${{ runner.arch }}"
+          echo "Working Directory: $(pwd)"
+          echo "User: $(whoami)"
+
+      - name: Test Docker
+        run: |
+          echo "Testing Docker availability..."
+          docker --version
+          docker ps
+
+      - name: Test AWS CLI
+        run: |
+          echo "AWS CLI Version:"
+          aws --version
+          echo ""
+          echo "AWS Region:"
+          echo $AWS_REGION
+          echo ""
+          echo "Caller Identity:"
+          aws sts get-caller-identity --output json
+
+      - name: Test Terraform
+        run: |
+          echo "Testing Terraform..."
+          terraform version
+
+      - name: Simple Build Test
+        run: |
+          echo "Running simple build test..."
+          cd test-module
+          echo "test" > test-file.txt
+          cat test-file.txt
+
+      - name: Summary
+        if: always()
+        run: |
+          echo "## Test Runner Validation Complete!" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "**Runner:** \`${{ runner.name }}\`" >> $GITHUB_STEP_SUMMARY
+          echo "**Branch:** \`${{ github.ref_name }}\`" >> $GITHUB_STEP_SUMMARY
+          echo "**Run ID:** \`${{ github.run_id }}\`" >> $GITHUB_STEP_SUMMARY
+          echo "**Run Attempt:** \`${{ github.run_attempt }}\`" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "### Capabilities Verified:" >> $GITHUB_STEP_SUMMARY
+          echo "- AWS credentials via IAM role" >> $GITHUB_STEP_SUMMARY
+          echo "- Docker support" >> $GITHUB_STEP_SUMMARY
+          echo "- Terraform CLI" >> $GITHUB_STEP_SUMMARY
+          echo "- Basic file operations" >> $GITHUB_STEP_SUMMARY
+```
+
+
+
+
+
+**Alternative: If the secret already exists**, you can use a data source and skip the targeted apply:
```

Comment:
It seems like this section should be part of Step 4 (before testing the setup with a workflow). Also, I think that the reference to a 'targeted apply' is maybe a leftover from a previous version? 

The validation script in `main.tf` recommends that the secret be created via the AWS CLI, and that's also what's documented here in step , and in the `github_secret_name` description in `variables.tf`. So my advice is to standardize on that approach. 

Right now it seems like you're also supporting the approach of managing the secret via Terraform (which is why you'd need `skip_webhook_creation` and a two-phase apply), but in my opinion that makes it confusing to use the module, since there's already so much documentation around creating the secret outside of Terraform (via the AWS CLI)